modules/govuk_env_sync/files/govuk_env_sync.sh

```diff
@@ -53,19 +53,44 @@ function set_filename {
 }
 
 function compress_data {
-  cd "${tempdir}"
-  tar --create --gzip --force-local --file "${filename}" "${database}"
+# We don't need compression in the case of elasticsearch
+# as the tool we use to dump/restore already handle it
+  if [ ${dbms} != 'elasticsearch' ]
+  then
+    cd "${tempdir}"
+    tar --create --gzip --force-local --file "${filename}" "${database}"
+  fi
 }
 
 function decompress_data {
-  cd "${tempdir}"
-  tar --extract --gzip --file "${filename}"
+# We don't need compression in the case of elasticsearch
+# as the tool we use to dump/restore already handle it
+  if [ ${dbms} != 'elasticsearch' ]
+  then
+    cd "${tempdir}"
+    tar --extract --gzip --file "${filename}"
+  fi
 }
 
 function is_writable_mongo {
   mongo --quiet --eval "print(db.isMaster()[\"ismaster\"]);" "localhost/$database"
 }
 
+function is_writable_elasticsearch {
+# We need a bit of black magic here unfortunately
+# We're not looking for a writable node, but rather a unique one to avoid restoring to every nodes
+# We arbitrary pick node one, but we cannot get this info from the cluster itself as it assigns random id to nodes
+# So we use the metadata from the AWS instance to get this
+  NODE_ID=$(curl -s http://169.254.169.254/latest/meta-data/iam/security-credentials | awk -F "-" '{print $NF}')
```

Comment:
I was curious as to what this was doing so tried it out... the URL needs a trailing slash :)

Reply:
ah crap, thanks!